src/pages/Debug/TransactionViolation/DebugTransactionViolationPage.tsx

```diff
@@ -0,0 +1,93 @@
+import type {StackScreenProps} from '@react-navigation/stack';
+import React, {useCallback, useMemo} from 'react';
+import {View} from 'react-native';
+import {useOnyx} from 'react-native-onyx';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import ScreenWrapper from '@components/ScreenWrapper';
+import TabSelector from '@components/TabSelector/TabSelector';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import Debug from '@libs/actions/Debug';
+import DebugUtils from '@libs/DebugUtils';
+import * as DeviceCapabilities from '@libs/DeviceCapabilities';
+import Navigation from '@libs/Navigation/Navigation';
+import OnyxTabNavigator, {TopTab} from '@libs/Navigation/OnyxTabNavigator';
+import type {DebugParamList} from '@libs/Navigation/types';
+import DebugDetails from '@pages/Debug/DebugDetails';
+import DebugJSON from '@pages/Debug/DebugJSON';
+import NotFoundPage from '@pages/ErrorPage/NotFoundPage';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type SCREENS from '@src/SCREENS';
+import type {TransactionViolation} from '@src/types/onyx';
+
+type DebugTransactionViolationPageProps = StackScreenProps<DebugParamList, typeof SCREENS.DEBUG.TRANSACTION_VIOLATION>;
+
+function DebugTransactionViolationPage({
+    route: {
+        params: {transactionID, index},
+    },
+}: DebugTransactionViolationPageProps) {
+    const {translate} = useLocalize();
+    const [transactionViolations] = useOnyx(`${ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS}${transactionID}`);
+    const transactionViolation = useMemo(() => transactionViolations?.[Number(index)], [index, transactionViolations]);
+    const styles = useThemeStyles();
+
+    const saveChanges = useCallback(
+        (data: Record<string, unknown>) => {
+            const updatedTransactionViolations = [...(transactionViolations ?? [])];
+            updatedTransactionViolations.splice(Number(index), 1, data as TransactionViolation);
+            Debug.setDebugData(`${ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS}${transactionID}`, updatedTransactionViolations);
+        },
+        [index, transactionID, transactionViolations],
+    );
+
+    const deleteTransactionViolation = useCallback(() => {
+        const updatedTransactionViolations = [...(transactionViolations ?? [])];
+        updatedTransactionViolations.splice(Number(index), 1);
+        Debug.setDebugData(`${ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS}${transactionID}`, updatedTransactionViolations);
```

Comment:
same above

Reply:
Initially I tried using that function but it wasn't saving the data properly, but not that I tried it again it seems to be working!

I think it's working now because I ended up doing some changes at the time, so I'm removing `Debug.setDebugData`